app/controllers/laboratory_samples_controller.rb

```diff
@@ -9,4 +9,26 @@ def index
     # paginate samples
     @samples = perform_pagination(@samples)
   end
+
+  def new
+    @sample = LaboratorySample.new
+  end
+
+  def create
+
+    @sample = LaboratorySample.new({:uuid => uuid_param})
```

Comment:
Never trust user input!

If the UUID can't be created by users (ie, the system will auto-generate them, as in this case), let's not read them from a user parameter.

In this case, a user can edit the DOM of the `/new` page to change the system-defined `uuid` to a custom one, and the app will accept that new string as a UUID - which will probably break many assumptions through the app.

Since the user shouldn't be able to edit the UUID, let's not allow them to set it. Either `save` the sample as soon as the user hits the "New sample" button, or, if we want to ask the user for extra info, let's ask that _without showing_ the UUID, and redirect users to the `show` page after creating the new Sample.